server/datastore/mysql/migrations/tables/20241122171434_RemoveDuplicateIndexes.go

```diff
@@ -0,0 +1,117 @@
+	// # fleet.migration_status_tables
+	// # ########################################################################
+	//
+	// # Uniqueness of id ignored because PRIMARY is a duplicate constraint
+	// # id is a duplicate of PRIMARY
+	// # Key definitions:
+	// #   UNIQUE KEY `id` (`id`)
+	// #   PRIMARY KEY (`id`),
+	// # Column types:
+	// #	  `id` bigint unsigned not null auto_increment
+	// # To remove this duplicate index, execute:
+	// ALTER TABLE `fleet`.`migration_status_tables` DROP INDEX `id`;
+	//
+	// # ########################################################################
+	// # fleet.policy_membership
+	// # ########################################################################
+	//
+	// # idx_policy_membership_policy_id is a left-prefix of PRIMARY
+	// # Key definitions:
+	// #   KEY `idx_policy_membership_policy_id` (`policy_id`),
+	// #   PRIMARY KEY (`policy_id`,`host_id`),
+	// # Column types:
+	// #	  `policy_id` int unsigned not null
+	// #	  `host_id` int unsigned not null
+	// # To remove this duplicate index, execute:
+	// ALTER TABLE `fleet`.`policy_membership` DROP INDEX `idx_policy_membership_policy_id`;
+	//
+	// # ########################################################################
+	// # fleet.software
+	// # ########################################################################
+	//
+	// # Key software_listing_idx ends with a prefix of the clustered index
+	// # Key definitions:
+	// #   KEY `software_listing_idx` (`name`,`id`),
+	// #   PRIMARY KEY (`id`),
+	// # Column types:
+	// #	  `name` varchar(255) collate utf8mb4_unicode_ci not null
+	// #	  `id` bigint unsigned not null auto_increment
+	// # To shorten this duplicate clustered index, execute:
+	// ALTER TABLE `fleet`.`software` DROP INDEX `software_listing_idx`, ADD INDEX `software_listing_idx` (`name`);
+	//
+	// # ########################################################################
+	// # fleet.software_cve
+	// # ########################################################################
+	//
+	// # software_cve_software_id is a left-prefix of unq_software_id_cve
+	// # Key definitions:
+	// #   KEY `software_cve_software_id` (`software_id`)
+	// #   UNIQUE KEY `unq_software_id_cve` (`software_id`,`cve`),
+	// # Column types:
+	// #	  `software_id` bigint unsigned default null
+	// #	  `cve` varchar(255) collate utf8mb4_unicode_ci not null
+	// # To remove this duplicate index, execute:
+	// ALTER TABLE `fleet`.`software_cve` DROP INDEX `software_cve_software_id`;
+
+	_, err := tx.Exec(
+		"ALTER TABLE `app_config_json` DROP INDEX `id`;" +
+			"ALTER TABLE `host_users` DROP INDEX `idx_uid_username`;" +
+			"ALTER TABLE `migration_status_tables` DROP INDEX `id`;" +
+			"ALTER TABLE `policy_membership` DROP INDEX `idx_policy_membership_policy_id`;" +
+			"ALTER TABLE `software` DROP INDEX `software_listing_idx`, ADD INDEX `software_listing_idx` (`name`);" +
+			"ALTER TABLE `software_cve` DROP INDEX `software_cve_software_id`;",
```

Comment:
I'm sure those changes all make sense given the usual quality of those percona tools, I'm just worried that some of those tables can be very big (`host_users` and the software ones esp.) and that statement could take a long time to complete (and run into some timeouts?). I vaguely remember us doing something for long migrations, but maybe it's ok to just call it out in the release notes, maybe @lucasmrod remembers something similar?

Reply:
Indeed, I too remember customers having issues with long running migrations (and therefore long offline time). Worth running the migration on a load test environment with ~100k hosts to have an estimate of offline time for customers.

Reply:
Yes, I believe this is our standard flow now for load testing. Set up the load test for the previous minor version, then run the migration. But I'll call it out in the QA section.